app/upgrades/signetlaunch/data_signed_fps_test.go

```diff
@@ -0,0 +1,109 @@
+package signetlaunch_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/babylonlabs-io/babylon/app"
+	v1 "github.com/babylonlabs-io/babylon/app/upgrades/signetlaunch"
+	btcstktypes "github.com/babylonlabs-io/babylon/x/btcstaking/types"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	txmodule "github.com/cosmos/cosmos-sdk/x/auth/tx/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateSignatureSignedFPsFromData(t *testing.T) {
```

Comment:
hmm we could probably also add validation if the string do not containan duplicates

Reply:
good idea, I will update

Reply:
done at https://github.com/babylonlabs-io/babylon/pull/28/commits/c503f3bf3ce59c673203f5b76b512c01f29397a4